ion/tokenizer.go

```diff
@@ -542,11 +542,11 @@ func (t *tokenizer) readString() (string, error) {
 		if err != nil {
 			return "", err
 		}
-
-		switch c {
-		case -1, '\n':
+		if isProhibitedControlChar(c) || c == '\n' {
```

Comment:
I think we should explicitly handle the `EOF`/`-1` case here and below. `isProhibitedControlChar` will catch it, but the method name makes me think it's only looking for [ASCII control characters](http://ascii-table.com/control-chars.php), which doesn't include -1.

Reply:
I see your point, but that means going back to [this comment](https://github.com/amzn/ion-go/pull/58#discussion_r437438308). 
We can either change the function name to something more generic like ```invalidStringCharacter``` or put the logic before [this commit](https://github.com/amzn/ion-go/pull/58/commits/43f7b28c8059f179535f48afc93137c6ef39869e) back.

Reply:
There's a subtle difference here. We've gone from:

```go
if isInvalidChar(c) {
    // Doesn't handle EOF, but could have based on the name
}
switch c {
case -1: // Handles EOF
```
to
```go
if isProhibitedControlChar(c) {
     // Handles EOF, but shouldn't based on the name.
}
```

I think `isProhibitedControlChar` is a more precise/communicative name, so I'd like to keep it. But that means we need the explicit EOF check:
```go
if c == -1 || c == '\n' || isProhibitedControlChar(c) {
    // Handles EOF
}
```

I think this is especially helpful since someone new to the codebase could reasonably expect EOF to be handled by the
```go
if err != nil {
// ...
}
```
above, since `err` is how the standard library reports `EOF`.

Reply:
Cool. Updated the pull request. Thanks.